Model/CreditmemoByOrderIncrementId.php

```diff
@@ -100,6 +102,35 @@ public function filterItemsToBeRefunded(Order $order, CreditmemoInterface $credi
         return $selectedItemsToRefund;
     }
 
+    /**
+     * @param Order $order
+     * @param CreditmemoInterface $creditmemo
+     * @param CreditmemoInterface $newCreditmemo
+     */
+    protected function addBackToStockStatus(Order $order, CreditmemoInterface $creditmemo, CreditmemoInterface $newCreditmemo)
+    {
+        $itemsToBackToStock = [];
+        /** @var \Magento\Sales\Model\Order\Item $orderItem */
+        foreach ($order->getAllItems() as $orderItem){
+            /** @var \Magento\Sales\Model\Order\Creditmemo\Item $inputItem */
+            foreach ($creditmemo->getItems() as $inputItem){
+                if (
+                    $orderItem->getSku() === $inputItem->getSku() &&
+                    $inputItem->getExtensionAttributes()->getBackToStock() &&
+                    (!in_array($inputItem->getSku(), $itemsToBackToStock))
```

Comment:
To help with readability could you refactor this logic into its own function the function would take the `$orderItem` the `$inputItem` and `$itemsToBack` as arguments